addons/pos_self_order/controllers/orders.py

```diff
@@ -232,13 +232,13 @@ def _process_lines(self, lines, pos_config, pos_order_id, take_away=False):
                     child_tax_results = child_selected_account_tax.compute_all(
                         child_price_unit,
                         pos_config.currency_id,
-                        child.get('qty'),
+                        1,
                         child_product,
                     )
                     newLines.append({
                         'price_unit': price_unit,
-                        'price_subtotal': child_tax_results.get('total_excluded'),
-                        'price_subtotal_incl': child_tax_results.get('total_included'),
+                        'price_subtotal': child_tax_results.get('total_excluded') * line.get('qty'),
+                        'price_subtotal_incl': child_tax_results.get('total_included') * line.get('qty'),
```

Comment:
Can't the rounding be different depending on the parameters and quantity? This is handled by _compute_all, I think.

@caburj can you confirm this or is it good like that ?

Reply:
Normally, compute_all should handle the calculation of the tax amount and pre-tax amount by giving it the price and quantity. I'd prefer it that way than computing the unit price ourselves and multiplying it to the quantity. With that, we are more consistent with the recalculation of taxes and pre-tax amounts during session closing, avoiding any account-move-posting-issues.

Nonetheless, the idea being proposed in this PR seems to be a good research to explore. If we prove that this is better than the classic way of computing tax-related values, we should applied it everywhere. But until we prove that we can do better with this, we'll keep the old implementation.